File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int size = 0;
        ListNode* temp = head;

        while(temp){
            size++;
            temp = temp -> next;
        }

        if(size == 1 || n > size) return NULL;

        n = size - n;
        temp = head;
        if(n == 0){
            head = head -> next;
            temp -> next = NULL;
            delete temp;
            return head;
        }

        for(int i = 1; i < n; i++){
            temp = temp -> next;
        }

        ListNode* toDel = temp -> next;
        temp -> next = temp -> next -> next;
        delete toDel;
        return head;
    }
};

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        size = 0
        temp = head 

        while temp : 
            size += 1
            temp = temp.next
        
        if size == 1 and n > size:
            return NULL
        
        n = size - n
        temp = head

        if n == 0:
            head = head.next
            return head
        
        for i in range(1,n):
            temp = temp.next
        
        temp.next = temp.next.next
        return head